src/braket/experimental/autoqasm/program/program.py

```diff
@@ -43,6 +44,10 @@ class UserConfig:
     """User-specified configurations that influence program building."""
 
     num_qubits: Optional[int] = None
+    """The total number of qubits to declare in the program."""
+
+    device: Optional[AwsDevice] = None
```

Comment:
Do you have an example of this? I actually wasn't aware of this scenario, would love to learn more about it.

Reply:
All I mean here is to make this `Optional[Device]` instead of `Optional[AwsDevice]`. `LocalSimulator` is not limited to the options we've defined (namely `LocalSimulator()`, `LocalSimulator("braket-dm")` and `LocalSimulator("braket-ahs")`), and in fact works with any customer-defined backend; see [this notebook](https://github.com/aws/amazon-braket-examples/blob/feature/cirq-plugin-using-program-context/examples/braket_features/create_custom_circuit_and_custom_local_simulator.ipynb) for a good example!

Reply:
Makes sense, that's very helpful - thank you! Updating this as follows: https://github.com/aws/amazon-braket-sdk-python/commit/b5efad3c820429f06fa112f7eb497ec41a531bd5